sw/device/silicon_creator/lib/drivers/spi_device.h

```diff
@@ -0,0 +1,60 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+#ifndef OPENTITAN_SW_DEVICE_SILICON_CREATOR_LIB_DRIVERS_SPI_DEVICE_H_
+#define OPENTITAN_SW_DEVICE_SILICON_CREATOR_LIB_DRIVERS_SPI_DEVICE_H_
+
+#include <stddef.h>
+#include <stdint.h>
+
+#include "sw/device/silicon_creator/lib/error.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#define SPI_DEVICE_WARN_UNUSED_RESULT __attribute__((warn_unused_result))
+
+enum {
+  /**
+   * lowRISC's manufacturer ID from the JEDEC JEP106BE standard.
+   */
+  kSpiDeviceJedecContCode = 0x7f,
+  kSpiDeviceJedecContCodeCount = 12,
+  kSpiDeviceJedecManufId = 0xef,
+  /**
+   * MSB of the 2-byte device ID.
+   */
+  kSpiDeviceJedecDevId = 0x07,
```

Comment:
If there is a hardware revision register somewhere that we'll increment with successive versions of the chip, we should use a value derived from that register here.

In my previous role, we used the upper nybble as the chip generation and the lower nybble as the chip revision:

```
CHIP_B0 = 0x00
CHIP_B1 = 0x01
CHIP_B2 = 0x02
CHIP_C0 = 0x10
CHIP_D0 = 0x20
CHIP_D2 = 0x22
```
May also want to consider reserving a bit to indicate whether this is ROM bootstrap mode vs. Owner Firmware mode.

Reply:
Thanks @cfrantz, created #11605 for the HW rev register.

For the device ID, I reserved a bit for the bootstrap mode, 3 bits for chip revision and 4 bits for chip generation, PTAL.

(mode & density will probably be arguments or moved to somewhere else but kept inline for now since owner firmware mode needs more discussion.)